chia/full_node/full_node_api.py

```diff
@@ -1295,7 +1324,9 @@ async def request_puzzle_solution(self, request: wallet_protocol.RequestPuzzleSo
 
         block_generator: Optional[BlockGenerator] = await self.full_node.blockchain.get_block_generator(block)
         assert block_generator is not None
-        error, puzzle, solution = get_puzzle_and_solution_for_coin(block_generator, coin_record.coin)
+        error, puzzle, solution = await asyncio.get_running_loop().run_in_executor(
+            self.executor, get_puzzle_and_solution_for_coin, block_generator, coin_record.coin
```

Comment:
Consider consistency with a partial here?  Even kwargs for all params in each case?  `:]`